From 382762d8a27b525563b41b8142d4cdc6bd164443 Mon Sep 17 00:00:00 2001
From: CvH <namerp@googlemail.com>
Date: Tue, 8 Aug 2017 17:21:56 +0200
Subject: [PATCH] fix

---
 drivers/media/usb/dvb-usb-v2/af9035.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/usb/dvb-usb-v2/af9035.c b/drivers/media/usb/dvb-usb-v2/af9035.c
index 4df9486..a2a2c1b 100644
--- a/drivers/media/usb/dvb-usb-v2/af9035.c
+++ b/drivers/media/usb/dvb-usb-v2/af9035.c
@@ -2124,6 +2124,10 @@ static const struct usb_device_id af9035_id_table[] = {
 	/* IT930x devices */
 	{ DVB_USB_DEVICE(USB_VID_ITETECH, USB_PID_ITETECH_IT9303,
 		&it930x_props, "ITE 9303 Generic", NULL) },
+	{ DVB_USB_DEVICE(USB_VID_DEXATEK, 0x0100,
+		&it930x_props, "Logilink VG0022A", NULL) },
+    { DVB_USB_DEVICE(USB_VID_TERRATEC, 0x10b2,
+        &it930x_props, "TerraTec Cinergy TC2 Stick", NULL) },
 	{ }
 };
 MODULE_DEVICE_TABLE(usb, af9035_id_table);
-- 
2.10.0.windows.1

